source/Octopus.Tentacle/Kubernetes/InMemoryTentacleScriptLog.cs

```diff
@@ -0,0 +1,40 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Octopus.Tentacle.Contracts;
+
+namespace Octopus.Tentacle.Kubernetes
+{
+    public class InMemoryTentacleScriptLog
+    {
+        readonly List<ProcessOutput> logs = new();
```

Comment:
Without being flip, is this not the appropriate place for a `ConcurrentDictionary`? It will handle all this locking for you so you don't need manual lock statements everywhere?

Reply:
We want to preserve the order of log messages right? Or are you saying this can be unordered since we merge this with the Pod logs anyway?

Reply:
I think we need a lock for `PopLogs()` anyway - otherwise we might clear the list when someone is adding